### DISCUSSION TITLE
[Enhancement] [table] 希望增加ElTable表格行数折叠功能

### DISCUSSION THREAD

**Discussion by XueHua-s** [element-plus/element-plus#13801] [Feature Request]
Feature type: **`Enhancement`**

## Details

- Enhanced component name: `table`

## Description

# 问题
关于Element plus中Table组件只有折叠内容，折叠表格树，并没有折叠行数这个功能。
## 需求
1.有些场景，表格可能没有分页。直接拉取的全部，比如某些功能的详情页。
2.在这种情况下，如果详情内表格过多，那么用户的体验肯定不要。例如: 需要滑动很远的距离，才能够看到自己想要的。
### 例子
自己通过Vue自定义指令，给表格实现的行数折叠。
http://www.mtfxh.wiki/2023/07/27/element-plus%E8%A1%A8%E6%A0%BC%E9%9A%90%E8%97%8F%E8%A1%8C%E6%95%B0%E5%8A%9F%E8%83%BD/

### Additional comments

    // css
    .el-folding > tr:nth-child(n + 4) {
	    display: none;
    }
    .el-folding-line {
	    transition-duration: 200ms;
	    cursor: pointer;
	    border-bottom: 1px solid var(--el-table-border-color);
    }
    .el-folding-line:active {
	    background-color: #F5F6FB;
    }
    .el-folding-extends {
	    /*cursor: pointer;*/
	    transition-duration: 400ms;
    }
    .el-folding-extends.rotate {
	    cursor: pointer;
	    transform: rotate(180deg);
    }
    
    // element-table折叠指令
      app.directive('elfold', (el, binding) => {
        const elBody = el.querySelector(".el-table__body>tbody")
        const elBodyC = elBody.children
        if (elBodyC.length > 3) {
          // 给表格添加只展示三行的类名
          if (el.querySelector('.rotate')) {
            elBody.classList.remove('el-folding')
          } else {
            elBody.classList.add('el-folding')
          }
          // 判断有没有创建过面板
          const foldingLine = el.querySelector('.el-folding-line')
          if (foldingLine) {
            if (el?.removeElement) {
              el.removeElement(foldingLine)
            }
          }
          if (!foldingLine) {
            // 创建折叠面板元素
            const extendsTop = document.createElement('div')
            extendsTop.className = "flex-ct p5 el-folding-line"
            const extendsTopButton = document.createElement('i')
            extendsTopButton.className = "fa fa-angle-down el-folding-extends"
            extendsTopButton.style.fontSize = "30px"
            extendsTop.append(extendsTopButton)
            el.append(extendsTop)
            // 控制变量
            let controller = true
            extendsTop.addEventListener('click', () => {
              if (controller) {
                extendsTopButton.classList.add('rotate')
                elBody.classList.remove('el-folding')
              } else {
                extendsTopButton.classList.remove('rotate')
                elBody.classList.add('el-folding')
              }
              controller = !controller
            })
          }
        } else {
          elBody.classList.remove('el-folding')
        }
      })

![img_3](https://github.com/element-plus/element-plus/assets/97370135/9d83ba4d-36f9-4053-8cd2-d7a10cba2bd0)
![img_4](https://github.com/element-plus/element-plus/assets/97370135/843d8df0-94a9-4b2a-81f1-00584d671ce8)

**Comment by 2350568089** [comment]
这个用select选择id，在选择完成时跳scollbar会更好一点吧
